### PR TITLE
detect and fix stms log field having epoch time ms for intercept plugins.

### LINKED ISSUE #7196: stms log field can have bad values.
Observed in ats80x and ats81x, master.

We tried to enable the stms field in our logs.  We started observing invalid values in some circumstances.

When the parent isn't visited the value is -1.  This isn't a valid value for a time field, it should be '0' since '0' time was spent contacting the origin.  stmsf in this case is set to '-.001', sts to '0' and stmsh to 'ffff...'

In the case an intercept plugin is in play (like the stats_over_http or slice plugins), the stms value is set to utc milliseconds (ie: 1600263600889).  I speculate that the stms field is initialized to some start time but never resolved with an end time later in the transaction.

### PR DESCRIPTION
**Pull request by traeak**
For intercept plugins like stats_over_http and slice the stms logging field may end up being set to epoch time in ms.
This fix detects that condition inside HttpSM by only allowing TS_MILESTONE_SERVER_CLOSE to be set if 
TS_MILESTONE_SERVER_CONNECT was already set.

LogAccess::marshal_server_resp_time_ms uses this code to calculate this stms field.
```
m_http_sm->milestones.difference_msec(TS_MILESTONE_SERVER_CONNECT, TS_MILESTONE_SERVER_CLOSE)
```

This is a mitigation in the code.

resolves #7196

### CODE REVIEW

**Comment by elsloo** [comment]
Hey @traeak, we observed this occurring in our environment due to requests handled by an intercept plugin. I reviewed the fix provided here in the context of running an intercept plugin, and I believe that it will resolve the issue. It appears that there's only one place where the `TS_MILESTONE_SERVER_CONNECT` milestone timing is set, inside `HttpSM::do_server_open(bool)`, and if that method is not called, the subsequent logging calculations that leverage this milestone produce unexpected results as you mentioned above.

I believe we should merge this PR. 👍

**Review by bneradt — DISMISSED** [review]

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -2976,7 +2976,10 @@ HttpSM::tunnel_handler_server(int event, HttpTunnelProducer *p)
 {
   STATE_ENTER(&HttpSM::tunnel_handler_server, event);
 
-  milestones[TS_MILESTONE_SERVER_CLOSE] = Thread::get_hrtime();
+  // The following fixes up an intercept handler which doesn't set TS_MILESTONE_SERVER_CONNECT by default.
```
Comment (bneradt) [inline proxy/http/HttpSM.cc:2979]: To clarify that this comment applies to the check itself rather than the whole conditional block, maybe reword as:

```
  // An intercept handler may not set TS_MILESTONE_SERVER_CONNECT by default. Therefore we only 
  // set TS_MILESTONE_SERVER_CLOSE if TS_MILESTONE_SERVER_CONNECT is set (non-zero), lest 
  // certain time statistics are calculated from epoch time.
```

Just a suggestion.

Comment (traeak) [reply proxy/http/HttpSM.cc]: updated PR with above comment.

**Review by randall — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Comment by zwoop** [comment]
Cherry-picked to 8.1.x
